engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
@vrozov Yes, I did see the JIRA which doesn't have the stack information you asked for. My comments were based on the JIRA and PR content which don't show the stack trace you have mentioned. In any case, I still don't understand the reason for caching nodeId and host if the root cause was that allocatedContainer was null.